script_mteb_french/estimate_evaluation_cost.py

```diff
@@ -0,0 +1,145 @@
+
+TASK_LIST_RETRIEVAL = [
+    "AlloprofRetrieval",
+    "BSARDRetrieval",
+    "HagridRetrieval",
+    "SyntecRetrieval",
+]
+
+TASK_LIST_STS = ["STSBenchmarkMultilingualSTS", "STS22"]
+
+TASK_LIST_SUMMARIZATION = [
+    "SummEvalFr",
+]
+
+TASK_LIST_BITEXTMINING = [
+    "DiaBLaBitextMining",
+    "FloresBitextMining",
+]
+
+
+TASK_LIST = (
+    TASK_LIST_CLASSIFICATION
+    + TASK_LIST_CLUSTERING
+    + TASK_LIST_PAIR_CLASSIFICATION
+    + TASK_LIST_RERANKING
+    + TASK_LIST_RETRIEVAL
+    + TASK_LIST_STS
+    + TASK_LIST_SUMMARIZATION
+    + TASK_LIST_BITEXTMINING
+)
+
+PRICE_PER_1K_TOKEN = 0.0001
+
+TASKS_LANGS = ["fr"]
+
+# TODO : add functions for cost estimation for other task types
+# TODO : add argparser
+# TODO: dump result in a file ?
+
+
+def estimate_cost_retrieval_task(
+    task: mteb.tasks.Retrieval, price_per_1k_tokens=0.0001
+):
+    """Evaluates the cost for a retrieval task
+
+    Args:
+        task (mteb.tasks.Retrieval): the task object of mteb
+        price_per_1k_tokens (float, optional): Price for 1000 tokens. Defaults to 0.0001.
+
+    Returns:
+        _type_: _description_
+    """
+    # we might not know which is the encoder for each model. So we use the base one from ADA
+    # it leads to approx 2.5 tokens/word (which is high), the the cost may be overestimated
+    tokenizer = tiktoken.get_encoding("cl100k_base")
+    task.load_data()
+    total_n_tokens = 0
+    for key in task.queries.keys():
+        queries_encodings_lengths = sum(
+            [len(tokenizer.encode(s)) for s in task.queries[key].values()]
+        )
+        documents_encodings_lengths = sum(
+            [len(tokenizer.encode(s["text"])) for s in task.corpus[key].values()]
+        )
+        total_n_tokens += queries_encodings_lengths + documents_encodings_lengths
+    cost = round(total_n_tokens / 1000 * price_per_1k_tokens, 2)
+    print(task.description["name"], ":", cost, "$")
+
+    return cost
+
+
+def estimate_cost_reranking_task(
+    task: mteb.tasks.Retrieval, price_per_1k_tokens=0.0001
+):
+    tokenizer = tiktoken.get_encoding("cl100k_base")
+    task.load_data()
+    encodings_lengths = task.dataset.map(
+        lambda x: {
+            "n_tokens": len(tokenizer.encode(x["query"]))
+            + sum([len(tokenizer.encode(s)) for s in x["positive"]])
+            + sum([len(tokenizer.encode(s)) for s in x["negative"]])
+        }
+    )
+    cost = round(
+        sum(encodings_lengths["test"]["n_tokens"]) / 1000 * price_per_1k_tokens, 2
+    )
+    print(task.description["name"], ":", cost, "$")
+
+    return cost
+
+
+evaluation = MTEB(tasks=TASK_LIST, task_langs=TASKS_LANGS)
+for task in evaluation.tasks:
+    print(task.description["name"])
+    match task.description["type"]:
+        case "Retrieval":
+            _ = estimate_cost_retrieval_task(task, PRICE_PER_1K_TOKEN)
+        case "Reranking":
+            _ = estimate_cost_reranking_task(task, PRICE_PER_1K_TOKEN)
```

Comment:
Would be nice to keep all values in a dict and commit it to a dataset repo in `lyon-nlp` HF (like mteb does for results here: https://huggingface.co/datasets/mteb/results/tree/main/results)

Reply:
True. Though as we don't know whant is the tokenizer used for models other than text-embedding-ada-002, the result of the calculation is not very precise. I was more ocnsidering to keep that as an esptimation for ourselves. But it could be shared